src/main/java/io/crate/client/jdbc/CrateConnection.java

```diff
@@ -333,22 +341,30 @@ public boolean isValid(int timeout) throws SQLException {
 
     @Override
     public void setClientInfo(String name, String value) throws SQLClientInfoException {
+        if (value != null) {
+            this.properties.setProperty(name, value);
+        } else {
+            this.properties.remove(name);
+        }
     }
 
     @Override
     public void setClientInfo(Properties properties) throws SQLClientInfoException {
+        if (properties.isEmpty()) {
+            this.properties.clear();
+        } else {
+            this.properties = properties;
+        }
     }
 
     @Override
     public String getClientInfo(String name) throws SQLException {
-        checkClosed();
-        return null;
+        return this.properties.getProperty(name);
```

Comment:
is it ok to return `null` if the property does not exists or should it throw an exception? pls check spec